### PR TITLE
MsGraph Files With Common API

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
reference: demisto/etc#20893
fixes: demisto/etc/issues/23943

## Description
MSGraph Files to packs
Implementing MicrosoftApiModule
Added support to authenticate using a self-deployed Azure application.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 1]
```diff
@@ -1,28 +1,18 @@
-import binascii
-
 import demistomock as demisto
 from CommonServerPython import *
 from CommonServerUserPython import *
-
-from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+# from Packs.ApiModules.Scripts.MicrosoftApiModule.MicrosoftApiModule import MicrosoftClient
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:4]: remove this comment

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 2]
```diff
@@ -815,35 +611,36 @@ def delete_file_command(client, args):
...
+
+        LOG(f"Command being called is {demisto.command()}")
+
         if demisto.command() == "test-module":
             # This is the call made when pressing the integration Test button.
             result = module_test(client)
             demisto.results(result)
-
         elif demisto.command() == "msgraph-delete-file":
             readable_output, raw_response = delete_file_command(client, demisto.args())
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:635]: `raw_response` is not used here

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/README.md [hunk 3]
```diff
@@ -55,10 +61,6 @@ After you successfully execute a command, a DBot message appears in the War Room
 
 Delete a DriveItem by using its ID
 
-##### Required Permissions
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/README.md:58]: it would be great if we could for each command mention the required permission, so it will be clear why we actually require the permissions you listed above
you can find it in microsoft api reference

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 4]
```diff
@@ -163,122 +103,25 @@ def url_validation(url):
     return url
 
 
-class Client(BaseClient):
+class MsGraphClient:
     """
-    Client will implement the service API, should not contain Demisto logic.
-    Should do requests and return data
+    Microsoft Graph Mail Client enables authorized access to a user's Office 365 mail data in a personal account.
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:108]: update docstring here

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 5]
```diff
@@ -298,23 +141,16 @@ def list_drives_in_site(self, site_id=None, limit=None, next_page_url=None):
                 "next_page_url: if you have used the limit argument."
             )
 
-        if limit:
-            params = urlencode({"$top": limit})
-        else:
-            params = ""
+        params = urlencode({"$top": limit}) if limit else ""
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:144]: please check if `urlencode` is actually needed here, i think we can remove it

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py (hunk 2 above)
Comment (Bargenish) [reply Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:635]: should I use:
` readable_output, _ = delete_file_command(client, demisto.args())`
or change the return value of that command?

Comment (Itay4) [reply Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:635]: change the line below to `return_outputs(readable_output=readable_output, raw_response=raw_response)`

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 6]
```diff
@@ -445,19 +260,13 @@ def download_file(self, object_type, object_type_id, item_id):
...
         """
         if object_type == "drives":
-            url = f"{object_type}/{object_type_id}/items/{item_id}/content"
+            uri = f"{object_type}/{object_type_id}/items/{item_id}/content"
 
         elif object_type in ["groups", "sites", "users"]:
-            url = f"{object_type}/{object_type_id}/drive/items/{item_id}/content"
+            uri = f"{object_type}/{object_type_id}/drive/items/{item_id}/content"
 
         # send request
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:268]:
```suggestion
```

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py [hunk 7]
```diff
@@ -469,53 +278,41 @@ def create_new_folder(self, object_type, object_type_id, parent_id, folder_name)
...
             "name": folder_name,
             "folder": {},
             "@microsoft.graph.conflictBehavior": "rename",
         }
-        self.headers["Content-Type"] = "application/json"
-        demisto.log(f"sending POST to {url}, with the next payload: {payload}")
-        return self.http_call(
-            "POST", full_url=url, json_data=payload, headers=self.headers, url_suffix=""
-        )
+        demisto.log(f"sending POST to {uri} suffix, with the next payload: {payload}")
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles.py:292]: lets remove that log, dont see why we need it

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py [hunk 8]
```diff
@@ -1,19 +1,19 @@
...
 import CommonServerPython
-from MicrosoftGraphFiles import epoch_seconds, get_encrypted, remove_identity_key, url_validation, \
-    parse_key_to_context, delete_file_command, download_file_command, list_sharepoint_sites_command, \
-    list_drive_content_command, create_new_folder_command, list_drives_in_site_command, Client
-
+from MicrosoftGraphFiles import remove_identity_key, url_validation, parse_key_to_context, delete_file_command, \
+    download_file_command, list_sharepoint_sites_command, list_drive_content_command, create_new_folder_command, \
+    list_drives_in_site_command, MsGraphClient
 
+# TODO: remove dots before paths
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:8]: remove TODO

File: Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py [hunk 9]
```diff
@@ -192,10 +160,10 @@ def test_parse_key_to_context_exclude_keys_from_list():
     "command, args, response, expected_result",
     [
         (
-            download_file_command,
-            {"object_type": "drives", "object_type_id": "123", "item_id": "232"},
-            File,
-            commands_expected_results["download_file"],
+                download_file_command,
```
Comment (Itay4) [inline Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:163]: linting here looks weird, why did you add the tab here?

Comment (Bargenish) [reply Packs/MicrosoftGraphFiles/Integrations/MicrosoftGraphFiles/MicrosoftGraphFiles_test.py:163]: I didn't. probably the reformat file and I haven't noticed. I see it elsewhere as well. I'll fix it.

**Comment by kirbles19** [comment]
@Bargenish Done.
